src/EditorFeatures/CSharpTest/ConvertToRecord/ConvertToRecordTests.cs

```diff
@@ -0,0 +1,864 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+using VerifyCS = Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions.CSharpCodeRefactoringVerifier<
+    Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertToRecord.CSharpConvertToRecordRefactoringProvider>;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertToRecord
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsConvertToRecord)]
+    public class ConvertToRecordTests
+    {
+        [Fact]
+        public async Task TestNoProperties_NoAction()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        // field, not property
+        public int f = 0;
+    }
+}
+";
+            await TestNoRefactoringAsync(initialMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestSetterProperty_NoAction()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; set; }
+    }
+}
+";
+            await TestNoRefactoringAsync(initialMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestSetterPropertyOnReadonlyStruct_NoAction()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public readonly struct [|C|]
+    {
+        public int P { get; }
+    }
+}
+";
+            await TestNoRefactoringAsync(initialMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestInitPropertyOnStruct_NoAction()
```

Comment:
i'm curious why.

Reply:
When `record struct`s have positional params, they are by default `get/set`. My idea is that we wouldn't want to change those semantics because they likely chose to make it `init` for a reason. However in a follow-up PR This will be converted to a positional param and then overridden (see explanation below)